frontend/src/components/Navigator/__tests__/index.js

```diff
@@ -61,4 +61,12 @@ describe('Navigator', () => {
     userEvent.click(screen.getByRole('button', { name: 'Continue' }));
     await waitFor(() => expect(onSubmit).toHaveBeenCalled());
   });
+
+  it('changes navigator state to complete when "continuing"', async () => {
+    renderNavigator();
+    userEvent.click(screen.getByRole('button', { name: 'Continue' }));
+    await screen.findByTestId('second');
+    const navItem = await screen.findByText('first page');
+    await waitFor(() => expect(within(navItem.nextSibling).getByText('Complete')).toBeVisible());
```

Comment:
This actually confused me more. Why is this first step complete when it was "in progress" during the other test?

Reply:
In the first test the user fills out some of the first page but never completes the first page by clicking the `Continue` button, they navigate via the side navigation to the second page. Pages are "Complete" only when the `Continue` button is pressed. The continue button is disabled unless the form is valid (all required fields have valid values).

In the second test the first page is valid because there are no required form fields, so the `Continue` button is active and sets the first page as "Complete" when clicked.

I decided against setting pages as "Complete" automatically when the form is valid. If we automatically set the form to valid after all required fields have valid values there is the potential for users to miss filling in optional fields.

Reply:
Ahhhh, ok.

I'm not 100% certain I like that limitation of only switching if you navigate with the button and not the nav menu, but I definitely see your point about optional fields so it's something we should address with more research and another story.